docs/proposals/20241127-predownload-in-vk-zh.md

```diff
@@ -0,0 +1,144 @@
+   * [Proposal](#proposal)
+      * [User Stories](#user-stories)
+         * [Serverless 场景](#serverless-场景)
+         * [混合部署弹性场景](#混合部署弹性场景)
+      * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+         * [如何指定调度到 vk](#如何指定调度到-vk)
+         * [指定预热镜像后，如何进行预热](#指定预热镜像后如何进行预热)
+         * [如何通过预热已有的 pod 来实现弹性池的预热呢？](#如何通过预热已有的-pod-来实现弹性池的预热呢)
+   * [Additional Details](#additional-details)
+      * [Test Plan [optional]](#test-plan-optional)
+   * [Implementation History](#implementation-history)
+
+## Summary
+
+镜像拉取作为加速pod启动的重要一环，我们之前已经支持了基于 kruise-daemon 调用 CRI 接口进行镜像预热的方案。但这种方案无法实现 virtual kubelet 场景下的镜像预热。
+我们注意到如 ACK 平台上 ECI 上已经支持了部分预热的方案，即对预热镜像的容器数据盘进行打快照来实现镜像加速。该提案旨在利用该机制实现 vk 场景下的镜像预热，并尽可能复用已有的 CR。
+
+## Motivation
+
+弹性节点或者 serverless 作为使用云的最大优势之一，其最大的优势在于弹性，如果能够将预热镜像的机制与弹性池的机制结合起来，将能够显著提升弹性池的弹性。
+
+### Goals
+
+- 尽可能复用已有 CR 的情况下支持预热vk
+- 支持用户自行指定预热镜像
+- 支持从已有pod上进行预热
+
+### Non-Goals/Future Work
+
+- 如果 vk本身不支持预热，本提案不负责实现
+
+## Proposal
+对于用户在 ImageListPullJob 指定的镜像，我们根据指定的镜像创建一个临时pod，并从该pod上进行预热。
+
+1. 增加一个字段标识是否需要在 vk 上进行预热
+计划在已有的 ImageListPullJob CR 上增加一个字段 `spec.predownloadInVK`，用于标识是否需要在 vk 上进行预热
+
+2. 增加一个字段标识预热 vk 的selector
+计划在 ImageListPullJob 中增加一个字段 `spec.vkConfig.selector`，用于标识预热 vk 的 selector, 该字段为可选字段，如果不指定则默认使用 `type = virtual-kubelet` 的 selector.
+
+```yaml
+apiVersion: apps.kruise.io/v1alpha1
+kind: ImageListPullJob
+metadata:
+  name: job-with-always
+spec:
+  images:
+  - nginx:1.9.1   # [required] image to pull
+  - busybox:1.29.2
+  - ...
+  predownloadInVK: true # new field
+  vkConfig:
+    selector: # [optional] if not assigned, defaults to type=virtual-kubelet
+      type: virtual-kubelet 
+      cloud: xxx
+  parallelism: 10      # [optional] the maximal number of Nodes that pull this image at the same time, defaults to 1
+  selector:            # [optional] the names or label selector to assign Nodes (only one of them can be set)
+    names:
+    - node-1
+    - node-2
+    matchLabels:
+      node-type: xxx
+  completionPolicy:
+    type: Always                  # [optional] defaults to Always
+    activeDeadlineSeconds: 1200   # [optional] no default, only work for Always type
+    ttlSecondsAfterFinished: 300  # [optional] no default, only work for Always type
+  pullPolicy:                     # [optional] defaults to backoffLimit=3, timeoutSeconds=600
+    backoffLimit: 3
+    timeoutSeconds: 300
+```
+
+### User Stories
+
+#### Serverless 场景
+
+用户可以在部署应用前预热镜像。虽然需要提前规划可能使用的镜像组合，但可以显著提升应用的启动速度。
+
+#### 混合部署弹性场景
+
+用户在常规资源池中运行应用，通过标记可能需要弹性扩容到 VK 的应用，Kruise 可以自动获取这些应用当前使用的镜像组合并进行预热。
+
+
+### Implementation Details/Notes/Constraints
+![alt text](../img/predownload-in-vk.png)
+在 ImageListPullJob 中定义了 vk 镜像预热的信息后，我们会根据是否需要预热 vk 场景，从而生成一个 job 指定调度到 vk 进行启动。
+#### 如何指定调度到 vk
+一般来说用户可以通过选择 `type = virtual-kubelet` 这个 label 来指定调度到 vk 上，但考虑到不同的云厂商实现，或者多个 vk 源的场景，我们可以增加一个字段让用户在特定场景下指定。
+
+#### 指定预热镜像后，如何进行预热
+参考阿里云 ECI 的预热方案，只要包含所有镜像的 pod 第一次成功启动，即可认为预热成功。
```

Comment:
it is not necessary to wait for the container to start successfully, as long as the container had started, the pull job should be considered as success, e.g. check the pod.status.containerstatus and to see the running and terminated status is nil or not